File: hw/chalan/u1/hw14Json/JavaScript/model/ChickenCoop.js
export default class ChickenCoop {
    #id;
    #farmerId;
    #chickens;

    constructor(id, farmerId) {
        this.#id = id;
        this.#farmerId = farmerId;
        this.#chickens = [];
    }

    getId() { return this.#id; }
    getFarmerId() { return this.#farmerId; }
    getChickens() { return this.#chickens; }

    addChicken(chicken) {
        this.#chickens.push(chicken);
    }

    toJSON() {
        return {
            id: this.#id,
            farmerId: this.#farmerId,
            chickens: this.#chickens.map(chicken => chicken.toJSON())
        };
    }

    static fromJSON(data) {
        const coop = new ChickenCoop(data.id, data.farmerId);
        
        // Importar Chicken de forma síncrona
        import('./Chicken.js').then(ChickenModule => {
            const Chicken = ChickenModule.default;
            data.chickens.forEach(chickenData => {
                coop.addChicken(Chicken.fromJSON(chickenData));
            });
        }).catch(error => {
            console.error("Error loading Chicken module:", error);
        });
        
        return coop;
    }

    toString() {
        const chickensInfo = this.#chickens.map(chicken => "  " + chicken.toString()).join("\n");
        return `ChickenCoop{id: ${this.#id}, farmerId: ${this.#farmerId}}\n${chickensInfo}`;
    }
}

File: hw/chalan/u1/hw14Json/python/model/chickenCoop.py
class ChickenCoop:
    def __init__(self, coop_id, farmer_id):
        self.coop_id = coop_id
        self.farmer_id = farmer_id  # ID del granjero dueño
        self.chickens = []  # Lista de objetos Chicken
    
    def add_chicken(self, chicken):
        self.chickens.append(chicken)
    
    def to_dict(self):
        return {
            'coop_id': self.coop_id,
            'farmer_id': self.farmer_id,
            'chickens': [chicken.to_dict() for chicken in self.chickens]
        }
    
    @classmethod
    def from_dict(cls, data):
        coop = cls(data['coop_id'], data['farmer_id'])
        from model.chicken import Chicken
        for chicken_data in data['chickens']:
            coop.add_chicken(Chicken.from_dict(chicken_data))
        return coop
    
    def __str__(self):
        chickens_info = "\n".join([f"  {chicken}" for chicken in self.chickens])
        return f"ChickenCoop(id={self.coop_id}, owner={self.farmer_id})\n{chickens_info}"